src/renderer/views/stake/AddStake/AddStakeView.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { useCallback } from 'react'
+
+import * as RD from '@devexperts/remote-data-ts'
+import { Asset, assetToString, bn, bnOrZero } from '@thorchain/asgardex-util'
+import * as FP from 'fp-ts/function'
+import * as A from 'fp-ts/lib/Array'
+import * as O from 'fp-ts/lib/Option'
+import { useObservableState } from 'observable-hooks'
+import { useHistory } from 'react-router'
+import { map } from 'rxjs/operators'
+
+import { AddStake } from '../../../components/stake/AddStake/AddStake'
+import { ONE_ASSET_BASE_AMOUNT, ZERO_BASE_AMOUNT, ZERO_BN } from '../../../const'
+import { useMidgardContext } from '../../../contexts/MidgardContext'
+import { useWalletContext } from '../../../contexts/WalletContext'
+import { getDefaultRuneAsset } from '../../../helpers/assetHelper'
+import { sequenceTRD } from '../../../helpers/fpHelpers'
+import { liveData } from '../../../helpers/rx/liveData'
+import * as stakeRoutes from '../../../routes/stake'
+import { PoolDetailRD } from '../../../services/midgard/types'
+import { getPoolDetail, toPoolData } from '../../../services/midgard/utils'
+import { getBalanceByAsset } from '../../../services/wallet/util'
+
+export const AddStakeView: React.FC<{ asset: Asset }> = ({ asset }) => {
+  const history = useHistory()
+
+  const onChangeAsset = useCallback(
+    (asset: Asset) => {
+      history.replace(stakeRoutes.stake.path({ asset: assetToString(asset) }))
+    },
+    [history]
+  )
+
+  const { service } = useMidgardContext()
```

Comment:
[nitpick] Pls desctruct `service.pools.xyz` from `service` to avoid repeating of `service.pools` on different places.